### DISCUSSION TITLE
Website and API in maintenances with important changes

### DISCUSSION THREAD

**Discussion by NYT92** [amvstrm/amvstrm#68] [Announcements]
Hello everyone, We want to tell you that our amvstrm's website and API will be in maintenance mode tomorrow because we are releasing a new update for the website and API. 

## Website
- Added a new control option for watch pages (Theater mode, Download, Settings, and Previous & Next function)
- Added Carousel to the mobile layouts
- Fixing the search page issue
- Fixing AnimeCard type error
- Fixing the PWA config issue

## API
- Rework the API from expressjs to elysiajs
- [Breaking Changes] ``/api/v2/stream`` will now have 2 parameters and will no longer accept the gogoanime episode id normally. 
  Example : ``/api/v2/stream/:gogoanime id/:episode number``
- [Breaking Changes] ``/api/v2/skiptime`` will be renamed to ``/api/v2/skiptimes``
- Added ``/swagger`` route for easy documentation
- Removed 9anime from ``id_providers``
- ``/api/v1/download`` will now provided raw URL for downloading anime
- Added hash option to hashed the IP address for rate-limiting
- Added ``GOGOANIME_PROXY`` and ``ANILIST_PROXY``
- Added custom provider to fix the incorrect provider or doesn't exist provider data with proper check

That is all we have done, We will provide more information to the update but for now, we will be putting everything in maintenance mode tomorrow. Thank you for reading the update

**Comment by devachandru** [comment]
Hey it's been more than 4 days and the api is still not working dude. Plz make the api live again.

**Reply by iotserver24** [reply]
Actually api is working but gogo has blocked on monday for all scrappers
Better use multiquality streaming player in settings of player
Its still working

**Reply by LeothosThoren** [reply]
Hi, thanks for your work, cheers 🥇 
I have a question though. Does this kind of blocking often happened? And it is related with your reworking?
Best regards.

**Reply by iotserver24** [reply]
No actually on this monday gogoanime planned and blocked all websites from scrapping it
Before the allow everyone CORs was enabled so everyone were scraping the site
Now as they disabled kt has become a huge issue right now

**Reply by NYT92** [reply]
Sorry for late response but It up now

**Reply by Josumaru** [reply]
Hii @NYT92 something wrong when hit episodes endpoint, is this problem still in maintenace phase?
![image](https://github.com/user-attachments/assets/6dabe957-76e2-470d-b04b-109cb8f04fc7)

**Reply by iotserver24** [reply]
@Josumaru no
Its because the gogoanime now blocked allow everyone CORS
So its giveing this issue

**Reply by Josumaru** [reply]
> Hii @NYT92 something wrong when hit episodes endpoint, is this problem still in maintenace phase? ![image](https://private-user-images.githubusercontent.com/114573843/386413787-6dabe957-76e2-470d-b04b-109cb8f04fc7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MzE2MzEsIm5iZiI6MTczMTYzMTMzMSwicGF0aCI6Ii8xMTQ1NzM4NDMvMzg2NDEzNzg3LTZkYWJlOTU3LTc2ZTItNDcwZC1iMDRiLTEwOWNiOGYwNGZjNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNVQwMDQyMTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ODFkZjBiZWM1N2M3YjNiMGI1ZjcwZTdiNWU3NWU4YmI3OWI1MWM4ZjBkMGFkODk1MDNkZjNhYWUwYTE3OGZhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.poo1NlM03CsFFF_Gq6BF0nW_drJiBxTK4mDjsCdZbjU)

may i use this [Prod API] instead of https://api.amvstr.me ?

**Reply by NYT92** [reply]
> Hii @NYT92 something wrong when hit episodes endpoint, is this problem still in maintenace phase?
> ![image](https://github.com/user-attachments/assets/6dabe957-76e2-470d-b04b-109cb8f04fc7)
> 
> 

The V2 relying on consumet and making the API slow also Anilist reduced the amount of the request (30/min). We will provide the proxy to the public API later.

And no, The production API only reserved for the amvstrm site.You can self-hosted yourself and we also open-source the API on the repo too.

**Reply by Josumaru** [reply]
> @Josumaru no Its because the gogoanime now blocked allow everyone CORS So its giveing this issue

hmm, ah i see

**Reply by Josumaru** [reply]
> > Hii @NYT92 something wrong when hit episodes endpoint, is this problem still in maintenace phase?
> > ![image](https://private-user-images.githubusercontent.com/114573843/386413787-6dabe957-76e2-470d-b04b-109cb8f04fc7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MzIzNzgsIm5iZiI6MTczMTYzMjA3OCwicGF0aCI6Ii8xMTQ1NzM4NDMvMzg2NDEzNzg3LTZkYWJlOTU3LTc2ZTItNDcwZC1iMDRiLTEwOWNiOGYwNGZjNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNVQwMDU0MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YjlmMGIyY2VjM2UyMzQ0MjdmODFkOWQ4ZTE4YWZmMTEyOWQ4YTk5ZThiYjM1ZjY0ZTk2OTNjYjg3ZWRhMTEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.jyTJuv6-1EIvJBuqEDRg0CWmElN0fBMffKjmuk8Ra0c)
> 
> The V2 relying on consumet and making the API slow also Anilist reduced the amount of the request (30/min). We will provide the proxy to the public API later.
> 
> And no, The production API only reserved for the amvstrm site.

okay, thank you sir

**Comment by Josumaru** [comment]
Hii sir, are the problem fix now?

**Reply by NYT92** [reply]
Yes but there might be issues sometimes

**Reply by Josumaru** [reply]
> Yes but there might be issues sometimes

for now i can't use v2/episodes for some ID, but in www3.amvstr.me work properly with v1 to get the episodes, are the problem for v2 will fix soon?

**Reply by NYT92** [reply]
V2 episode is still relying on consumet but we plan to implement that on our own soon we just still haven't figured it out yet...

**Comment by sandipansingh** [comment]
@Josumaru In the meantime, take a look at our API here: https://github.com/animesage-online/animesage-api

**Reply by Josumaru** [reply]
> @Josumaru In the meantime, take a look at our API here: https://github.com/animesage-online/animesage-api

great

**Reply by iotserver24** [reply]
I don't think this will work
Coz the issue is from gogoanime not from amvstrm api
Please try to understand that they have Disabled ALLOW ALL CORS
So u need to try to scrape in diff way

**Reply by sandipansingh** [reply]
@iotserver24 bro the stream soures are working for me in both  amvstrm api as well as in animesage-api

**Reply by iotserver24** [reply]
But its not working for me as well as to @NYT92

**Reply by iotserver24** [reply]
Thats the actual reason for this delay

**Reply by sandipansingh** [reply]
@iotserver24 can you tell me in which part you are getting the error and the error response??
mail it to me at animesage@animesage.online

**Comment by rishabnotfound** [comment]


**Reply by NYT92** [reply]
It does the job but it is too limited that are set by the owner. It could work but some ``.ts`` stream files might exceed more than 5mb sometimes causing the stream error and the proxy only allows ``5 per second`` so I don't think it will work in the long run...
 
![image](https://github.com/user-attachments/assets/714f49d2-93a8-45b8-8421-e3e434d4ba7c)

**Comment by rishabnotfound** [comment]
And at this point
GogoAnime fuck3d us lmao

Now the whole api is not working
Rip to every gogoanime scrapper

**Reply by sandipansingh** [reply]
https://api.animesage.online/

and for endpoint documentation go to https://docs.animesage.online/


Use this

**Comment by LeothosThoren** [comment]
Hi @NYT92,
I have noticed since a couple of weeks that it might be an issue with endpoints using Path or Query parameters. I always get  an http error code 522 due to time out connection. It is an issue already known? 
The issue occur with both V1 and V2 apis. For example this endpoint works without query params https://api.amvstr.me/api/v2/popular. The host name seems to work too.
<img width="321" alt="Capture d’écran 2025-01-12 à 15 52 52" src="https://github.com/user-attachments/assets/e1483522-70c1-4fbf-b5c9-3b9a8e169b27" />
<img width="1174" alt="Capture d’écran 2025-01-12 à 15 53 24" src="https://github.com/user-attachments/assets/26f37b72-d393-4c57-b776-30628a07849f" />

Thanks a lot for your help :)

**Reply by NYT92** [reply]
We have a problem with the server right now and We will tell everyone once it goes back to normal.

**Reply by LeothosThoren** [reply]
Allright thanks !👌
Will you communicate on Github or do you have a specific channel to do so?
